sparta/sparta/utils/SpartaSharedPointer.hpp

```diff
@@ -50,46 +71,28 @@ namespace sparta
     template <class PointerT>
     class SpartaSharedPointer
     {
+    public:
+        class SpartaWeakPointer;
+
+    private:
         /// Internal structure to keep track of the reference
         struct RefCount
         {
             explicit RefCount(PointerT * _p,
                               bool perform_delete = true) :
-                count(1),
                 p(_p),
-                perform_delete_(perform_delete){}
+                perform_delete(perform_delete)
+            {}
 
-            ~RefCount() {
-                if (perform_delete_) {
-                    delete p;
-                    p = nullptr;
-                }
-            }
+            ~RefCount() { p = nullptr; }
```

Comment:
Is `perform_delete` no longer used?

Reply:
Oops. That's a bug.  Good catch!

Reply:
This bug is pretty deep -- I forgot decrement the wp_count on assignment/move assignment on the original cnt before replacing.  Removing this variable exposed that.  Thanks!